### ISSUE TITLE
Detect a non-compatible browser?

### ISSUE THREAD

**Issue by tomek-he-him** [Jam3/tap-dev-tool#6]
Hi, I need to run my CSS tests in IE. It looks like they don’t support some standard for styled output. (What a surprise!):

![](https://gist.githubusercontent.com/tomekwi/d41ba242ddb7f07cc29a/raw/0d0cd26f58cdb9a4d2acc5e825ac080287673f82/shot.png)

Is it possible to detect if that standard is supported?

**Comment by tomek-he-him** [comment]
(BTW, the test results are also quite funny :smiley:)

**Comment by mattdesl** [comment]
Hmm I have a couple ideas but maybe there is a specific API for detecting this. I will ask around.

**Comment by tomek-he-him** [comment]
:+1: Thanks!

**Comment by mattdesl** [comment]
Actually it looks like there is no standard, and some browser detection might be the way to go. e.g.
https://github.com/icodeforlove/Console.js/blob/master/src/Console.js#L12

We can find a detect function that works and wrap that up as a separate module. :smile: Seems like latest Chrome, FF, and Safari all support CSS in console.

**Comment by mattdesl** [comment]
Also re: test results, do you think it's tap-dev-tool or just the browser rounding stuff weirdly?

**Comment by tomek-he-him** [comment]
>  do you think it's tap-dev-tool or just the browser rounding stuff weirdly?

It’s the browser. It’s the result of some floats + [`.getBoundingClientRect()`](http://devdocs.io/dom/element/getboundingclientrect). Chrome and FF are OK with it, but IE seems to get confused.

Sorry for not being explicit.

**Comment by tomek-he-him** [comment]
> find a detect function that works and wrap that up as a separate module.

Good idea! Even if sniffing is unavoidable it’d be good to have that abstracted away.

**Comment by tomek-he-him** [comment]
BTW, https://github.com/icodeforlove/Console.js itself looks good – perhaps using that for printing could save us some work?

**Comment by tomek-he-him** [comment]
It does [modify the prototype of `String`](https://github.com/icodeforlove/Console.js/tree/aa9bcc8747a77a2825a2de4588b82f09bb142d75#modern-usage-chrome--firefox--ie--9) – that could bite someone if used inside a library like this one. But there is a [compatibility mode](https://github.com/icodeforlove/Console.js/tree/aa9bcc8747a77a2825a2de4588b82f09bb142d75#compatibility-usage) which doesn’t to it.
